src/tree.c

```diff
@@ -81,6 +81,30 @@ int git_tree_entry_icmp(const git_tree_entry *e1, const git_tree_entry *e2)
 		git__strncasecmp);
 }
 
+/**
+ * Allocate a tree entry, borrowing the filename from the tree which
+ * owns it. This is useful when reading trees, so we don't allocate a
+ * ton of small strings but can use the pool.
+ */
+static git_tree_entry *alloc_entry_pooled(git_pool *pool, const char *filename, size_t filename_len)
+{
+	git_tree_entry *entry = NULL;
+	size_t tree_len;
+
```

Comment:
Can we add a check here to make sure that `filename_len` fits in a `uint16_t`?  I don't see any obvious vulnerabilities if one were to somehow build a tree that had a longer filename in it, but I'm also not the most creative person in the owrld.